validation/mock_metric_validation.go

```diff
@@ -0,0 +1,97 @@
+package validation
+
+import (
+	"fmt"
+	"reflect"
+
+	"google.golang.org/genproto/googleapis/monitoring/v3"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+)
+
+const (
+	missingFieldMsg = "%v must contain requried %v field"
+)
+
+func IsValidGetMetricDescriptorRequest(req *monitoring.GetMetricDescriptorRequest) error {
```

Comment:
Not sure if this would work but can you abstract all of these by having them take in an `interface{}`? Since you're only using `req` to call `reflect.ValueOf(req)`, and everything else is the same

Reply:
yeah definitely